creds/vault/manager.go

```diff
@@ -17,6 +18,8 @@ type VaultManager struct {
 
 	PathPrefix string `long:"path-prefix" default:"/concourse" description:"Path under which to namespace credential lookup."`
 
+	Cache bool `bool:"cache" default:"false" description:"Cache returned secrets for their lease duration in memory"`
```

Comment:
Should we also add a adjustable value for secret.LeaseDuration - so people can adjust to their usecase? E.g in our case, those secrets are 99% read only and if we write, we would restart ATC. People mighr use vault differently with AWS and rotation, so setting this down a lot ( or disable the cache ). But being able to define the general aggressiveness would be a way to help people adjusting to their use-case very easy, with very little effort on our side. What do you think?